src/types/TypeUtils.ts

```diff
@@ -0,0 +1 @@
+export type InsureBasis<T extends Y, Y = any> = T;
```

Comment:
"Insure" comes from the word "insurance", it does partially fit, but I think that "ensure" will be better.
Either way I don't think "InsureBasis" or "EnsureBasis" are meaningful enough, the type name should **clearly** reflect it's meaning,
Since it constrains a type by another how about `Constrained`
```suggestion
export type Constrained<T extends Y, Y = any> = T;
```